Update CI

ClFeSc opened the pull request:
Since Windows CI runs don't actually work (the output of the CI is empty, and the CI "passes" when tests are known to fail (e.g. [here](https://github.com/hpi-swa-teaching/SketchMorph2/pull/64/checks?check_run_id=2705528009))), I thought it would be good to remove them from running the CI.

DevSchmidtchen commented:
Would it be possible to make them working like they should? Are other repositories working with those tests?

ClFeSc commented:
> Would it be possible to make them working like they should? Are other repositories working with those tests?

Good point. I looked at Algernon. They seem to have fixed this. I'll try it.

ClFeSc commented:
Ok, I completely changed the target of this PR: Windows now functions properly. I also removed the `pull_request` trigger for the CI since the checks run for all pushes. And I changed `smalltalk-version` to `smalltalk-image` since the CI complained about `smalltalk-version` being deprecated.
